app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -12,7 +12,7 @@ module HostManagedExtensions
         alias_method_chain :smart_proxy_ids, :katello
 
         has_one :content_host, :class_name => "Katello::System", :foreign_key => :host_id,
-                               :dependent => :restrict, :inverse_of => :foreman_host
+                               :dependent => :restrict_with_error, :inverse_of => :foreman_host
```

Comment:
I see a bunch of these changes (from `restrict` to `restrict_with_error`) but I think we want `restrict_with_exception` as it mirrors the behavior of `restrict` in Rails 3.